client/state/preferences/reducer.js

```diff
@@ -72,6 +72,16 @@ export const fetching = ( state = false, action ) => {
 	return state;
 };
 
+export const saving = ( state = false, action ) => {
+	switch ( action.type ) {
+		case PREFERENCES_SET:
+			return true;
+		case PREFERENCES_SAVE_SUCCESS:
+			return false;
```

Comment:
We should handle the network failure case too to make sure the boolean doesn't get stuck `true` forever.

```suggestion
		case PREFERENCES_SAVE_SUCCESS:
		case PREFERENCES_SAVE_FAILURE:
			return false;
```

Take a look at the `PREFERENCES_FETCH_FAILURE` action to see where it's dispatched.